mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -138,48 +137,151 @@ static Value applyTransforms(PatternRewriter &b, ThreadwiseWriteAllOp storeOp,
   return alloc;
 }
 
-static Operation *traceToNonViewOp(Operation *op) {
+static void traceToNonViewUsers(Operation *op,
+                                SmallVectorImpl<Operation *> &nonViewOps) {
+  if (auto transform = dyn_cast<TransformOp>(op)) {
+    Value result = transform.getResult();
+    for (auto &use : result.getUses()) {
+      traceToNonViewUsers(use.getOwner(), nonViewOps);
+    }
+  } else {
+    nonViewOps.push_back(op);
+  }
+}
+
+// This function traces to non view readers.
+// TODO(@manupak): we should really be implementing
+// MemoryEffectOpInterface to ops to make this simpler
+// to identify readers.
+static LogicalResult
+traceToNonViewReaders(Operation *op, Value parentVal,
+                      SmallVectorImpl<Operation *> &nonViewReaders) {
   if (auto transform = dyn_cast<TransformOp>(op)) {
```

Comment:
I'll just comment here that this `transform` care shouldn't be firing - the regularization pass ought to eliminate transforms on readers of things written by, say, a threadwise_write_all or la.generic.